make_level.py

```diff
@@ -0,0 +1,128 @@
+import sys
+import os
+
+cwd = os.getcwd()
+
+def write_init(skill_name, skill_path, level_name):
+    # If skills/skill/__init__.py doesn't exist, create a basic version (no levels)
+    if not os.path.exists(os.path.join(skill_path,"__init__.py")):
+        with open(os.path.join(skill_path,"__init__.py"), 'w+') as fp:
+            level_setup = "\n".join([
+                "import pkg_resources\n",
+                "from gitgud.skills.util import BasicLevel",
+                "from gitgud.skills.util import Skill",
+                "",
+                "skill = Skill(",
+                "    '{}',".format(skill_name),
+                "    [",
+                "    ]",
+                ")",
+                ""
+                ])
+            fp.write(level_setup)
+    
+        
+        # Read file
+        with open(os.path.join("gitgud","skills","__init__.py"), 'r') as fp:
+            filedata = fp.read()
+
+        # Add import statement
+        replace = "\n".join([
+                            "from gitgud.skills.{0} import skill as {0}_skill".format(skill_name),
+                            "",
+                            "",
+                            "from gitgud.skills.u"
+                            ])
+        filedata = filedata.replace("\nfrom gitgud.skills.u", replace)
+        
+        # Add to input array of Skill
+        replace = ",\n    {}_skill\n]".format(skill_name)
+        filedata = filedata.replace("\n]", replace)
+
+        # Write to file
+        filepath = os.path.join("gitgud","skills","__init__.py")
+        with open(filepath, 'w') as fp:
+            fp.write(filedata)
+            print("Registered skill \"{}\" in {}".format(skill_name, filepath))
+        
+        write_init(skill_name, skill_path, level_name)
+        return
+    else:
+        # Populate file with BasicLevel
+        with open(os.path.join(skill_path,"__init__.py"), 'r') as fp:
+            filedata = fp.read()
+
+        replace = ",\n        BasicLevel('{0}', pkg_resources.resource_filename(__name__, '_{0}/'))\n    ]".format(level_name)
+        filedata = filedata.replace("\n    ]", replace)
+        filedata = filedata.replace("[,","[")
+        
+        filepath = os.path.join(skill_path,"__init__.py")
+        with open(filepath, 'w') as fp:
+            fp.write(filedata)
+            print("Registered level \"{}\" in {}".format(level_name, filepath))
+    return
+
+def create_level_file(level_path,filename):
+    filepath = os.path.join(level_path,filename)
+    with open(filepath,'a+'):
+        pass
+    print("Created: {}".format(filepath))
```

Comment:
Let's make one file each for each and use them as templates for each of the files. The templates should explain what's going on and still work, so the .spec files should be valid (use tag and branch names to imply what each thing does), goal.txt should be short, and instructions.txt should use >>> and explain itself on multiple lines.

Once you do that, have this function copy those files over to the appropriate directory. It might make sense to look for a copy function in the os module.

Reply:
Done